src/mpid/ch4/src/ch4_init.h

```diff
@@ -290,19 +290,25 @@ MPL_STATIC_INLINE_PREFIX int MPID_Init(int *argc,
     }
 #endif
 
+    {
+        int shm_tag_ub = MPIR_Process.attrs.tag_ub, nm_tag_ub = MPIR_Process.attrs.tag_ub;
```

Comment:
In the current code, this line is zero initializing the tag_ub values. In a `--disable-ch4-shm` configuration, this will cause us to have an overall tag_ub of 0. We might want to consider grabbing https://github.com/pmodels/mpich/pull/2984/commits/787a648c3958212db1fab681a529e6de31b94b06 and https://github.com/pmodels/mpich/pull/2984/commits/136880219f3524905355dad962feeda50b148411 from my other PR to fix initialization.